### PR TITLE
Added some missing UT

### PR DESCRIPTION
**Pull request by emanuelmetal**
## Background
We need to update test suite to handle code coverage, increase coverage and output results to files

## changes
Updated all components tests
Uptaded all services tests
Updated jest configuration to output results to file and collect coverage
Added eslint result file

### CODE REVIEW

File: webapp/app/components/Main.test.js [hunk 1]
```diff
@@ -1,19 +1,50 @@
...
+import { Main, mapDispatchToProps, mapStateToProps } from './Main'; // Undecorated version.
+import initialState from '../reducers/initialState';
 
 function setup(props) {
   return shallow(<Main {...props} />);
 }
 
 describe('<Main /> component', () => {
   it('renders itself', () => {
+    // Arrange
```
Comment (javier-pepe) [inline webapp/app/components/Main.test.js:13]: Please remove the commented out code, if comments are to be commited, we should make sure they serve a purpose, i.e. to explain a non obvious solution, as a general rule we avoid obvious comments to not pollute the code.

File: webapp/app/components/Main.test.js [hunk 2]
```diff
@@ -1,19 +1,50 @@
...
     const wrapper = setup({
       actions: {},
       usersActions: {}
     });
 
+    // Assert
```
Comment (javier-pepe) [inline webapp/app/components/Main.test.js:19]: Please remove the commented out code, if comments are to be commited, we should make sure they serve a purpose, i.e. to explain a non obvious solution, as a general rule we avoid obvious comments to not pollute the code.

File: webapp/app/components/Main.test.js [hunk 3]
```diff
@@ -1,19 +1,50 @@
...
     expect(wrapper.find('Switch')).toHaveLength(1);
     expect(wrapper.find('Route')).toHaveLength(3);
   });
+
+  describe('mapToProps functions', () => {
```
Comment (javier-pepe) [inline webapp/app/components/Main.test.js]: `mapToProps` should be `mapStateToProps` function and a separate describe block for `mapDispatchToProps`

File: webapp/app/components/Main.test.js [hunk 4]
```diff
@@ -1,19 +1,50 @@
...
+    it('should return the state of auth module', () => {
+      // Act
+      const props = mapStateToProps(initialState);
+
+      // Assert
```
Comment (javier-pepe) [inline webapp/app/components/Main.test.js]: Please remove the commented out code, if comments are to be commited, we should make sure they serve a purpose, i.e. to explain a non obvious solution, as a general rule we avoid obvious comments to not pollute the code.

File: webapp/app/components/Main.test.js [hunk 5]
```diff
@@ -1,19 +1,50 @@
...
+      expect(props.isAuthenticated).toBe(false);
+    });
+
+    it('should return the binded actions ', () => {
+      // Arrange
```
Comment (javier-pepe) [inline webapp/app/components/Main.test.js]: Please remove the commented out code, if comments are to be commited, we should make sure they serve a purpose, i.e. to explain a non obvious solution, as a general rule we avoid obvious comments to not pollute the code.

File: webapp/app/components/Main.test.js [hunk 6]
```diff
@@ -1,19 +1,50 @@
...
+      const dispatch = () => {};
+      const expectedActions = [
+        'loginRequest',
+        'loginSuccess',
+        'loginError',
+        'login'
+      ];
+      // Act
```
Comment (javier-pepe) [inline webapp/app/components/Main.test.js]: Please remove the commented out code, if comments are to be commited, we should make sure they serve a purpose, i.e. to explain a non obvious solution, as a general rule we avoid obvious comments to not pollute the code.

**Review by javier-pepe — CHANGES_REQUESTED** [review]

File: webapp/app/components/Main.test.js (hunk 6 above)
Comment (emanuelmetal) [reply webapp/app/components/Main.test.js]: I think is a good practice to keep tests split in three parts, see [ArrangeActAssert](http://wiki.c2.com/?ArrangeActAssert), if this project will be used as startup point for other developer this comments shall remain to keep tests patter organized

@mravinale thougths?

Comment (mravinale) [reply webapp/app/components/Main.test.js]: Both have good arguments, but grouping in "arrange act and assert" helps a lot on reading, but if we agree to do this we should be consistent and update all tests.

personally I like more to have comments regarding "arrange act and assert", but is your choice as a team

**Review by mravinale — APPROVED** [review]

**Pull request merged**